Source/SantaGUI/SNTNotificationManager.m

```diff
@@ -143,4 +165,74 @@ - (void)postRuleSyncNotificationWithCustomMessage:(NSString *)message {
   [[NSUserNotificationCenter defaultUserNotificationCenter] deliverNotification:un];
 }
 
+#pragma mark SNTBundleNotifierXPC protocol methods
+
+- (void)updateTotalFileCountForEvent:(SNTStoredEvent *)event
+                         binaryCount:(uint64_t)binaryCount
+                           fileCount:(uint64_t)fileCount {
+  if ([self.currentWindowController.event.idx isEqual:event.idx]) {
+    dispatch_async(dispatch_get_main_queue(), ^{
+      self.currentWindowController.foundFileCountLabel.stringValue =
+          [NSString stringWithFormat:@"%llu binaries / %llu files", binaryCount, fileCount];
+    });
+  }
+}
+
+- (void)setBundleServiceListener:(NSXPCListenerEndpoint *)listener {
+  SNTXPCConnection *c = [[SNTXPCConnection alloc] initClientWithListener:listener];
+  c.remoteInterface = [SNTXPCBundleServiceInterface bundleServiceInterface];
+  [c resume];
+  self.bundleServiceConnection = c;
+  dispatch_semaphore_signal(self.bundleServiceSema);
+}
+
+#pragma mark SNTBundleNotifierXPC helper methods
+
+- (void)hashBundleBinariesForEvent:(SNTStoredEvent *)event {
+  // Wait until the bundle service is connected
+  dispatch_semaphore_wait(self.bundleServiceSema, DISPATCH_TIME_FOREVER);
```

Comment:
Yeah, 6 secs to allow the bundle service to reconnect is reasonable. Otherwise abandon bundle hashing and display blockable event.
Done.